fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyUtils.java

```diff
@@ -0,0 +1,120 @@
+        LocalDate overdueSinceDate = null;
+        CollectionData collectionData = CollectionData.template();
+        BigDecimal amountAvailable = BigDecimal.ZERO;
+        BigDecimal outstandingAmount = BigDecimal.ZERO;
+        boolean oldestOverdueInstallment = false;
+        boolean overdueSinceDateWasSet = false;
+        boolean firstNotYetDueInstallment = false;
+
+        // Get the oldest overdue installment if exists one
+        for (LoanRepaymentScheduleInstallment installment : loan.getRepaymentScheduleInstallments()) {
+            if (!installment.isObligationsMet()) {
+                if (installment.getDueDate().isBefore(businessDate)) {
+                    outstandingAmount = outstandingAmount.add(installment.getTotalOutstanding(loanCurrency).getAmount());
+                    if (!oldestOverdueInstallment) {
+                        oldestOverdueInstallment = true;
+                        overdueSinceDate = installment.getDueDate();
+                        overdueSinceDateWasSet = true;
+
+                        amountAvailable = installment.getTotalPaid(loanCurrency).getAmount();
+
+                        for (LoanTransactionToRepaymentScheduleMapping mappngInstallment : installment
+                                .getLoanTransactionToRepaymentScheduleMappings()) {
+                            final LoanTransaction loanTransaction = mappngInstallment.getLoanTransaction();
+                            if (loanTransaction.isChargeback()) {
+                                amountAvailable = amountAvailable.subtract(loanTransaction.getAmount());
+                                if (amountAvailable.compareTo(BigDecimal.ZERO) < 0) {
+                                    overdueSinceDate = loanTransaction.getTransactionDate();
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+            } else if (!firstNotYetDueInstallment) {
+                firstNotYetDueInstallment = true;
+                amountAvailable = installment.getTotalPaid(loanCurrency).getAmount();
+
+                for (LoanTransactionToRepaymentScheduleMapping mappngInstallment : installment
+                        .getLoanTransactionToRepaymentScheduleMappings()) {
+                    final LoanTransaction loanTransaction = mappngInstallment.getLoanTransaction();
+                    if (loanTransaction.isChargeback() && loanTransaction.getTransactionDate().isBefore(businessDate)) {
+                        amountAvailable = amountAvailable.subtract(loanTransaction.getAmount());
+                        if (amountAvailable.compareTo(BigDecimal.ZERO) < 0 && !overdueSinceDateWasSet) {
+                            overdueSinceDate = loanTransaction.getTransactionDate();
+                            overdueSinceDateWasSet = true;
+                            break;
```

Comment:
We dont want to break the iteration. We need to process all the due chargeback transactions. Please remove the `break`.

Reply:
Removed